src/services/codefixes/importFixes.ts

```diff
@@ -521,29 +521,29 @@ namespace ts.codefix {
         currentTokenMeaning: SemanticMeaning,
         cancellationToken: CancellationToken,
         sourceFile: SourceFile,
-        checker: TypeChecker,
         program: Program,
         useAutoImportProvider: boolean,
         host: LanguageServiceHost
     ): ReadonlyESMap<string, readonly SymbolExportInfo[]> {
         // For each original symbol, keep all re-exports of that symbol together so we can call `getCodeActionsForImport` on the whole group at once.
         // Maps symbol id to info for modules providing that symbol (original export + re-exports).
         const originalSymbolToExportInfos = createMultiMap<SymbolExportInfo>();
-        function addSymbol(moduleSymbol: Symbol, exportedSymbol: Symbol, importKind: ImportKind): void {
+        function addSymbol(moduleSymbol: Symbol, exportedSymbol: Symbol, importKind: ImportKind, checker: TypeChecker): void {
             originalSymbolToExportInfos.add(getUniqueSymbolId(exportedSymbol, checker).toString(), { moduleSymbol, importKind, exportedSymbolIsTypeOnly: isTypeOnlySymbol(exportedSymbol, checker) });
         }
-        forEachExternalModuleToImportFrom(program, host, sourceFile, /*filterByPackageJson*/ true, useAutoImportProvider, moduleSymbol => {
+        forEachExternalModuleToImportFrom(program, host, sourceFile, /*filterByPackageJson*/ true, useAutoImportProvider, (moduleSymbol, _, program) => {
+            const checker = program.getTypeChecker();

```

Comment:
When there’s an AutoImportProviderProject, these modules can come from different programs, but this was always asking for exports of module symbols with the same checker instance, which would be missing resolved exports cached on symbol links if the module was checked by a different checker. That’s why a `program` is passed to this callback, but I missed this usage of it in the original refactor.